MantidQt/CustomInterfaces/test/TomographyIfaceModelTest.h

```diff
@@ -221,13 +225,225 @@ class TomographyIfaceModelTest : public CxxTest::TestSuite {
+    // the custom one just processes a single member
+    std::vector<std::string> inputArgsVector{
+        "--some params --some other params"};
+
+    TestableTomographyIfaceModel model;
+
+    std::string inputArgsString =
+        model.constructSingleStringFromVector(inputArgsVector);
+
+    std::shared_ptr<TomoRecToolConfig> d = std::shared_ptr<TomoRecToolConfig>(
+        new ToolConfigCustom(inputRunnable, inputArgsString));
+
+    model.usingTool(TestableTomographyIfaceModel::g_customCmdTool);
+    model.setCurrentToolMethod("gridrec");
+
+    model.setCurrentToolSettings(d);
+
+    const std::string resource = TestableTomographyIfaceModel::g_SCARFName;
+
+    std::string actualRun;
+    std::vector<std::string> actualArgsVector;
+    model.makeRunnableWithOptions(resource, actualRun, actualArgsVector);
+
+    std::string expectedRunnable = "/scriptPath/";
+    // the space at the end is necessary, because of how
+    // constructSingleStringFromVector works
+    std::vector<std::string> expectedArgsVector{
+        "--some params --some other params "};
+    TS_ASSERT_EQUALS(actualRun, expectedRunnable);
+    TS_ASSERT_EQUALS(expectedArgsVector.size(), actualArgsVector.size());
+
+    for (size_t i = 0; i < expectedArgsVector.size(); ++i) {
+      // append the whitespace because it is added in the argument separation
+      TS_ASSERT_EQUALS(expectedArgsVector[i], actualArgsVector[i]);
+    }
+  }
+
+  void test_makeLocalRunnableWithOptionsCustom() {
+    std::string inputRunnable = "python /scriptPath/";
+    // the custom one just processes a single member
+    std::vector<std::string> inputArgsVector{
+        "--some params --some other params"};
+
+    TestableTomographyIfaceModel model;
+
+    std::string inputArgsString =
+        model.constructSingleStringFromVector(inputArgsVector);
+
+    std::shared_ptr<TomoRecToolConfig> d = std::shared_ptr<TomoRecToolConfig>(
+        new ToolConfigCustom(inputRunnable, inputArgsString));
+
+    model.usingTool(TestableTomographyIfaceModel::g_customCmdTool);
+    model.setCurrentToolMethod("gridrec");
+
+    model.setCurrentToolSettings(d);
+
+    const std::string resource = model.localComputeResource();
+
+    std::string actualRun;
+    std::vector<std::string> actualArgsVector;
+    model.makeRunnableWithOptions(resource, actualRun, actualArgsVector);
+
+    std::string expectedRunnable = "python";
+    // the space at the end is necessary, because of how
+    // constructSingleStringFromVector works
+    std::vector<std::string> expectedArgsVector{
+        "/scriptPath/ --some params --some other params "};
+    TS_ASSERT_EQUALS(actualRun, expectedRunnable);
+    TS_ASSERT_EQUALS(expectedArgsVector.size(), actualArgsVector.size());
+
+    for (size_t i = 0; i < expectedArgsVector.size(); ++i) {
+      // append the whitespace because it is added in the argument separation
+      TS_ASSERT_EQUALS(expectedArgsVector[i], actualArgsVector[i]);
```

Comment:
The for loop can be removed and turned into:
`TS_ASSERT(expectedArgsVector==actualArgsVector);`
See [here](http://stackoverflow.com/a/16422594)